async-nats/src/auth.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2020-2022 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#[derive(Clone)]
+pub(crate) enum AuthStyle {
+    /// No authentication.
+    NoAuth,
+
+    /// Authenticate using a token.
+    Token(String),
+
+    /// Authenticate using a username and password.
+    UserPass(String, String),
```

Comment:
Do we want to abbreviate? I don't think we're consistent with this.

Reply:
no, will un-abbreviate. It should nto be `Credentials` tho, as that can be easily confused with `.creds` file.